debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -253,7 +253,7 @@ public SchemaBuilder schemaBuilder(Column column) {
             case PgOid.INT4RANGE_ARRAY:
             case PgOid.NUM_RANGE_ARRAY:
             case PgOid.INT8RANGE_ARRAY:
-                return SchemaBuilder.array(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
+                return SchemaBuilder.array(SchemaBuilder.STRING_SCHEMA);
```

Comment:
Let's say the DDL operation was 
`ALTER TABLE CUSTOMERS ADD COLUMN col2 text[] DEFAULT '{a,b}'::text[] NOT NULL;`
So how it works is when the avro schema is created from the connect schema, The default value {"a","b"} for col2 is proposed to the avro schema. It verifies this default value. [Refer this.](https://github.com/apache/avro/blob/release-1.11.3/lang/java/avro/src/main/java/org/apache/avro/Schema.java#L1674-L1675) Now since we have kept the type optional so this verification fails since the string "a" is not a null, which is the first type in the array of types in this union.

Having said that, i also see that there's a [fix for this](https://issues.apache.org/jira/browse/AVRO-3649) which has been merged 5 months before, but there has been no release after that.

Reply:
@akanimesh7 OK, TBH I am really not fan of this change if it just by-pass a bug in Avro. To handle this properly we there should be an SMT developed that will flip the schema optionality if needed but I don't think this should go to the core.

Reply:
@jpechane Understood. In that case rather than doing it for all array types, can we just do it for text[] and varchar[] ? As soon as the newer version of avro is released, these two also can be corrected. Even earlier these two types were not supported, hence there is no harm, rather this is just an addition.

On the other hand creation of SMT will need some more time and will also be temporary code that will come in.

Reply:
Well, the problem is that you focus on it from Avro PoV only. But there are also converters for JSON and Protobuf. It does not feel correct to me to emit incorrect schema type for everybody to fix issue in Avro only.

Reply:
Okay, makes sense. Let's halt this one then until a new version of avro comes out with the bug fix.